pkg/ovsdb/transact.go

```diff
@@ -0,0 +1,711 @@
+	}
+	if fn == FN_LT && actual < expected {
+		return true, nil
+	}
+	if fn == FN_LE && actual <= expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func condBoolean(act interface{}, fn string, exp interface{}) (bool, error) {
+	actual, ok := act.(bool)
+	if !ok {
+		return false, errors.New("failed to convert to boolean")
+	}
+	expected, ok := exp.(bool)
+	if !ok {
+		return false, errors.New("failed to convert to boolean")
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func condString(act interface{}, fn string, exp interface{}) (bool, error) {
+	actual, ok := act.(string)
+	if !ok {
+		return false, errors.New("failed to convert to string")
+	}
+	expected, ok := exp.(string)
+	if !ok {
+		return false, errors.New("failed to convert to string")
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func condUUID(act interface{}, fn string, exp interface{}) (bool, error) {
+	actual, ok := act.(libovsdb.UUID)
+	if !ok {
+		return false, errors.New("failed to convert to uuid")
+	}
+	expected, ok := exp.(libovsdb.UUID)
+	if !ok {
+		return false, errors.New("failed to convert to uuid")
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func isEqualSet(a, b libovsdb.OvsSet) bool {
+	return true // FIXME: missing deep comparison
+}
+
+func isEqualMap(a, b libovsdb.OvsMap) bool {
+	return true // FIXME: missing deep comparison
+}
+
+func condSet(act interface{}, fn string, exp interface{}) (bool, error) {
+	actual, ok := act.(libovsdb.OvsSet)
+	if !ok {
+		return false, errors.New("failed to convert to set")
+	}
+	expected, ok := exp.(libovsdb.OvsSet)
+	if !ok {
+		return false, errors.New("failed to convert to set")
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && isEqualSet(actual, expected) {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && !isEqualSet(actual, expected) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func condMap(act interface{}, fn string, exp interface{}) (bool, error) {
+	actual, ok := act.(libovsdb.OvsMap)
+	if !ok {
+		return false, errors.New("failed to convert to map")
+	}
+	expected, ok := exp.(libovsdb.OvsMap)
+	if !ok {
+		return false, errors.New("failed to convert to map")
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && isEqualMap(actual, expected) {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && !isEqualMap(actual, expected) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func condTypeSwitch(actual interface{}, fn string, expected interface{}) (bool, error) {
+	switch actual.(type) {
+	case int:
+		return condInteger(actual, fn, expected)
+	case float64:
```

Comment:
float32 ?

Reply:
no. this is how Real is interpreted (I think I saw this in libovsdb that real is float64)